.github/workflows/test-backend.yml

```diff
@@ -0,0 +1,47 @@
+name: Test Backend
+on: 
+  pull_request:
+
+jobs:
+  testing:
+    name: Testing Backend
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        working-directory: ./backend
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Setup Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: '3.12'
+
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install -r requirements.txt
+          pip install pytest-md pytest-emoji
+
+        # some tests require an .env file to run.
+        # file should contain all the required properties.  
```

Comment:
oh interesting, should we be mocking the config for all tests so wouldn't need this?

Reply:
probably 😄 

Without the environment file, 7 tests fail. One example is the neo4j driver throwing errors as it needed a valid URL to initialise - although the tests don't seem to actually use the driver.

Since this could require a bit of refactoring, would you be happy if I investigated/fixed in a separate PR?

Reply:
yeah of course, I reckon a separate ticket to investigate / refactor sounds great